src/components/Class/EditClass.js

```diff
@@ -0,0 +1,315 @@
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setShowModal(false);
+    setDeleteCohort(false);
+  };
+
+  const handleEdit = () => {
+    setEditShowModal(true);
+    setAnchorElUser(null);
+  };
+
+  const handleCloseEdit = (e) => {
+    e.stopPropagation();
+    setEditShowModal(false);
+  };
+
+  const handleClickOpen = () => {
+    setShowModal(!showModal);
+
+    setAnchorElUser(null);
+  };
+
+  const handleClickOpenUnenroll = () => {
+    setunenrollShowModal(!unenrollShowModal);
+
+    setAnchorElUser(null);
+  };
+
+  const rolesList = user.data.user.rolesList;
+  let flag = false;
+  rolesList.includes("admin") || rolesList.includes("classAdmin")
+    ? (flag = true)
+    : (flag = false);
+
+  // API CALL FOR DELETE CLASS
+  const deleteHandler = (id) => {
+    const notify = () => {
+      toast.success(" Deleted the class successfully", {
+        position: toast.POSITION.BOTTOM_RIGHT,
+        autoClose: 2500,
+      });
+    };
+    setShowModal(!showModal);
+    return axios({
+      method: METHODS.DELETE,
+      url: `${process.env.REACT_APP_MERAKI_URL}/classes/${id}`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+        "delete-all": indicator,
+      },
+    }).then(() => {
+      setRefreshKey(true);
+      notify();
+      dispatch(classActions.deleteClass(id));
+    });
+  };
+
+  const ACBPathway = Newpathways?.find((path) => {
+    return item?.PartnerSpecificBatches?.pathway_id === path.id;
+  });
+
+  useEffect(() => {
+    if (classRefresh) {
+      dispatch(classActions.getClasses());
+      setClassRefresh(false);
+    }
+  }, [dispatch]);
+
+  const { data = [] } = useSelector(({ Class }) => Class.allClasses);
+
+  const pathwayFilter =
+    data &&
+    data?.filter((data) => {
+      // if I click on a pathway, batches should be filter by that pathway
+      return data?.pathway_id === item?.PartnerSpecificBatches?.pathway_id;
+    });
+
+  return (
+    <>
+      <MoreVertIcon
+        style={{ color: "#BDBDBD", cursor: "pointer" }}
+        onClick={handleOpenUserMenu}
+        sx={{ p: 0 }}
+      />
+
+      {/* dialog box for edit delete and merge class  */}
+      <Menu
+        sx={{ mt: "15px" }}
+        id="menu-appbar"
+        anchorEl={anchorElUser}
+        anchorOrigin={{
+          vertical: "top",
+          horizontal: "right",
+        }}
+        keepMounted
+        maxWidth="130px"
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "right",
+        }}
+        PaperProps={{
+          style: { width: "150px" },
+        }}
+        open={Boolean(anchorElUser)}
+        onClose={() => {
+          setAnchorElUser(null);
+        }}
+      >
+        {(item?.facilitator?.email === user.data.user.email || flag) && (
+          <>
+            <MenuItem
+              onClick={(e) => {
+                e.stopPropagation();
+                handleEdit(item.id);
```

Comment:
`handleEdit` doesn't take any inputs.
```suggestion
                handleEdit();
```